### ISSUE TITLE
Component fails to package with ng-packagr

### ISSUE THREAD

**Issue by ddklo** [kuhnroyal/mat-datetimepicker#33]
When testing the component in an Angular 6 library I get the following error when running ng-packagr:

> Directive MatDatetimepickerInput, Property '_dateAdapter' is private and only accessible within class 'MatDatetimepickerInput<D>'

I think it might be related to AoT and the use of the property in a directive with host.

**Comment by kuhnroyal** [comment]
I can fix this specific error but I am not sure what else is needed to get it running.

**Comment by kuhnroyal** [comment]
I have update the project to latest angular CLI and ng-packagr. The sample app runs fine with `ng serve --prod` which should have AOT enabled. I am not sure why my build is not catching this.

**Comment by kuhnroyal** [comment]
You can try with `2.0.1-beta.0`

**Comment by ddklo** [comment]
Thank you for updating, but I still get the same error with 2.0.1-beta.0.  `ng serve --prod` works fine for me too. It is running `ng-packagr -p ng-package.json` that fails. Where ng-package.json contains the following:

```json
{
  "$schema": "./node_modules/ng-packagr/ng-package.schema.json",
  "lib": {
    "entryFile": "public_api.ts",
    "languageLevel": ["dom", "es2015", "es2016", "es2017"]
  },
  "whitelistedNonPeerDependencies": ["."],
  "dest": "./.package/common"
}
```

ng-packagr is version 3.0.3

**Comment by kuhnroyal** [comment]
I didn't fix the problem yet, I want to catch this in my own build first.
Do you use a custom tsconfig?

**Comment by kuhnroyal** [comment]
I published `2.0.1-beta.1` in which I made the `_dateAdapter` property public.

**Comment by ddklo** [comment]
Thanks, beta.1 fixed the issue for me.

tsconfig is almost standard (from angular cli 5.x project), but there are a few options added to support generating the lib.

``` json
{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": true,
    "module": "es2015",
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2017",
      "dom"
    ],
    "baseUrl": "src",
  },
  "angularCompilerOptions": {
    "strictMetadataEmit": true,
    "skipTemplateCodegen": true,
    "preserveWhitespaces": false
  }
}
```

**Comment by kuhnroyal** [comment]
Thanks, I still can't get my build to fail tho.

**Comment by kuhnroyal** [comment]
Released in 2.0.1
